### DISCUSSION TITLE
[Support] Unable to connect to LAN if WG connected on LAN

### DISCUSSION THREAD

**Discussion by Rouzax** [pivpn/pivpn#1377] [Support] [answered]
## In raising this issue, I confirm the following:

`{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pivpn/pivpn/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [] The issue I am reporting can be *is* directly related to the pivpn installer script.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

### Issue
I have PiVPN with a WireGuard setup running on a RPi4 in combination with PiHole. 
Have forwarded the port to the RPi and all works as intended when I connect from outside my LAN to WG.
Will get a connection and am able to browse internet and connect to internal resources on my LAN.

Problems arise when I'm connected to my LAN (either wifi or cabled) and try to still have that WG connection.
(The idea is that the WG is always on to make sure all traffic routes through home as this will keep PiHole active)
When connected to the LAN and an active WG connection I can still internet but no internal LAN resources are reachable.

When doing a ping I get `1  General failure.`

### Have you searched for similar issues and solutions?
    (yes/no / which issues?)
yes

### Console output of      `curl -L install.pivpn.io | bash`
```
  Output Here
```

### Console output of      `pivpn add` or `pivpn add nopass`
```
  Output Here
```

### Console output of      `pivpn debug`
```

pi@pihole:~ $ pivpn debug
::: Generating Debug Output
::::            PiVPN debug              ::::
=============================================
::::            Latest commit            ::::
commit 5aa0de636f32fd674b77a2b3dd64d97b51680707
Author: 4s3ti <4s3ti@protonmail.com>
Date:   Mon May 31 21:01:46 2021 +0200

    IRC From freenode to libera.chat

    Changed info about IRC Chanel,
    Chanel moved from freenode to libera.chat
=============================================
::::        Installation settings        ::::
PLAT=Raspbian
OSCN=buster
USING_UFW=0
IPv4dev=eth0
dhcpReserv=1
IPv4addr=192.168.2.18/24
IPv4gw=192.168.2.254
install_user=pi
install_home=/home/pi
VPN=wireguard
pivpnPORT=51820
pivpnDNS1=10.6.0.1
pivpnDNS2=
pivpnHOST=REDACTED
INPUT_CHAIN_EDITED=0
FORWARD_CHAIN_EDITED=0
pivpnPROTO=udp
pivpnMTU=1400
pivpnDEV=wg0
pivpnNET=10.6.0.0
subnetClass=24
ALLOWED_IPS="0.0.0.0/0, ::0/0"
UNATTUPG=1
INSTALLED_PACKAGES=(wireguard-tools qrencode unattended-upgrades)
=============================================
::::  Server configuration shown below   ::::
[Interface]
PrivateKey = server_priv
Address = 10.6.0.1/24
MTU = 1420
ListenPort = 51820
### begin USER1-S20FE ###
[Peer]
PublicKey = USER1-S20FE_pub
PresharedKey = USER1-S20FE_psk
AllowedIPs = 10.6.0.2/32
### end USER1-S20FE ###
### begin USER1-Corp-Laptop ###
[Peer]
PublicKey = USER1-Corp-Laptop_pub
PresharedKey = USER1-Corp-Laptop_psk
AllowedIPs = 10.6.0.3/32
### end USER1-Corp-Laptop ###
### begin USER2 ###
[Peer]
PublicKey = USER2_pub
PresharedKey = USER2_psk
AllowedIPs = 10.6.0.4/32
### end USER2 ###
=============================================
::::  Client configuration shown below   ::::
[Interface]
PrivateKey = USER1-S20FE_priv
Address = 10.6.0.2/24
MTU = 1400
DNS = 10.6.0.1

[Peer]
PublicKey = server_pub
PresharedKey = USER1-S20FE_psk
Endpoint = REDACTED:51820
AllowedIPs = 0.0.0.0/0, ::0/0
=============================================
::::    Recursive list of files in       ::::
::::    /etc/wireguard shown below       ::::
/etc/wireguard:
configs
keys
wg0.conf

/etc/wireguard/configs:
clients.txt
USER1-Corp-Laptop.conf
USER1-Corp-Laptop.conf.save
USER1-S20FE.conf
USER2.conf

/etc/wireguard/keys:
USER1-Corp-Laptop_priv
USER1-Corp-Laptop_psk
USER1-Corp-Laptop_pub
USER1-S20FE_priv
USER1-S20FE_psk
USER1-S20FE_pub
USER2_priv
USER2_psk
USER2_pub
server_priv
server_pub
=============================================
::::            Self check               ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] WireGuard is running
:: [OK] WireGuard is enabled (it will automatically start on reboot)
:: [OK] WireGuard is listening on port 51820/udp
=============================================
:::: Having trouble connecting? Take a look at the FAQ:
:::: https://github.com/pivpn/pivpn/wiki/FAQ
=============================================
:::: WARNING: This script should have automatically masked sensitive       ::::
:::: information, however, still make sure that PrivateKey, PublicKey      ::::
:::: and PresharedKey are masked before reporting an issue. An example key ::::
:::: that you should NOT see in this log looks like this:                  ::::
:::: YIAoJVsdIeyvXfGGDDadHh6AxsMRymZTnnzZoAb9cxRe                          ::::
=============================================
::::            Debug complete           ::::
:::
::: Debug output completed above.
::: Copy saved to /tmp/debug.log

```
### Have you taken any steps towards solving your issue?
While connected to LAN and WG, tracert to internet
```
tracert -w 100 -d google.com

Tracing route to google.com [142.250.179.206]
over a maximum of 30 hops:

  1     6 ms     2 ms     1 ms  10.6.0.1
  2     1 ms     1 ms     1 ms  192.168.2.254
  3    12 ms     3 ms     2 ms  195.190.228.75
  4     *        *        *     Request timed out.
  5     *        *        *     Request timed out.
  6     4 ms     4 ms     4 ms  142.250.238.171
  7     5 ms     5 ms     5 ms  142.251.48.179
  8     4 ms     3 ms     4 ms  142.250.179.206

Trace complete.
```

While connected to LAN and WG, tracert to internal resource
```
tracert -w 100 -d 192.168.2.10

Tracing route to 192.168.2.10 over a maximum of 30 hops

  1  General failure.

Trace complete.
```

It is likely a routing issue but can't figure out where it breaks down

**Comment by coolapso** [comment, marked as answer]
Hi @Rouzax 

> It is likely a routing issue but can't figure out where it breaks down

you are entirely right there, and the reason is very simple, that is not how the vpn is intended to be used! 

If you have VPN inside your home, you connect from outside into home! 
connecting to the VPN when you are already inside your home is pointless and useless. 

Read this, hope it helps you: https://blog.4s3ti.net/vpn-where-to.html

**Reply by Rouzax** [reply]
I understand the intent of VPN and the way it functions
Is there then anyway to have WireGuard Android client suspend if Wi-Fi network x is connected, for instance?

**Reply by orazioedoardo** [reply]
You might be able to do it with Tasker: https://www.youtube.com/watch?v=pCldPHn5SBg

**Reply by Rouzax** [reply]
Great idea

**Comment by coolapso** [comment]
Moving this to discussions

**Comment by woifes** [comment]
Ok it might be not useful but it should work right? I used it that way for years but all of a sudden it stopped working overnight. Why did it work for me in the first place?
